src-self-hosted/type.zig

```diff
@@ -708,6 +746,16 @@ pub const Type = extern union {
 
                 return std.math.ceilPowerOfTwoPromote(u16, (bits + 7) / 8);
             },
+
+            .optional => {
```

Comment:
This logic either needs to handle the possibility that the child type is a pointer (and therefore should have a different representation and therefore ABI size) or it should assert that the child type is not a pointer (and we will have to uphold this invariant in outside code). My suggestion would be to handle the possibility here, and allow any valid type for the child type.

Note that stage2 does not have canonical representations of types - the different forms are there to save memory, and to allow concurrent manipulation of types without synchronization.